### ISSUE TITLE
Type error on hiking import

### ISSUE THREAD

**Issue by hholzgra** [waymarkedtrails/waymarked-trails-site#358]
With last and this weeks planet file hiking import fails with:

```
./makedb.py -d planet hiking import
[...]
sqlalchemy.exc.DataError: (psycopg2.errors.InvalidParameterValue) Geometry type (GeometryCollection) does not match column type (LineString)

[SQL: INSERT INTO hiking.segments (rels, nodes, ways, geom) VALUES (%(rels)s, %(nodes)s, %(ways)s, ST_GeomFromEWKT(%(geom)s)) RETURNING hiking.segments.id]
[parameters: {'rels': [2381, 171840, 188265, 10188660], 'nodes': [70755227, 70755230, 70755232, 70755233, 70755237, 540244025, 70755239], 'ways': [9351839], 'geom': 'SRID=3857;GEOMETRYCOLLECTION EMPTY'}]
[...]
```

There wasn't a problem when I did a full planet import at end of May, with the planet file published on May 25th.

(The same problem occurs when using the waymarkedtrails-backend repository, but issue tracking isn't enabled on that repository yet?)

**Comment by hholzgra** [comment]
Thinking again there's also another difference to the import in May: that one was on Ubuntu 19.10 with PostgreSQL 11.7 and PostGIS 2.5.3 while the current attempt was on Ubuntu 20.04LTS with PostgreSQL 12.2 and PostGIS 3.0.0 ...

SqlAlchemy installed from PIP, not from Ubuntu repository, old system in May using 1.3.11, current one 1.3.17

**Comment by lonvia** [comment]
Odd, this looks like your node geometries are missing. The flatnode file is in order?

I have not yet tried the whole thing with Postgis 3. It might break things. I'd ask you to try with postgis 2 but I'm aware that this is hard with ubuntu 20.04. :(

NB: I've now enabled issues for `waymarkedtrails-backed`.

**Comment by hholzgra** [comment]
I've been using 20.04 for some time now with smaller extracts without problems. It's the first time I tried with a full planet though. Import doesn't fail immediately, but only after several minutes. 

Node cache file looks fine, it is minimally smaller than the osm2pgsql one, but that one already had some days of minutely diffs applied. Comparing the first few pages of hex dump the two files look pretty much the same.

I'll try again now with the planet file from late May, to see whether that also fails with newer software versions, but will probably not have results for that before tomorrow ...

**Comment by lonvia** [comment]
Best strategy for narrowing this down is probably to narrow down the input data and see if it is reproducible. I'd try with an extract that contains only relations 2381, 171840, 188265, 10188660 and their members.

I have a current planet on waymarkedtrails.org but it got to its state through updates so might not have hit the particular situation.

**Comment by hholzgra** [comment]
Same problem with the dump from May 25th that was imported just fine on an Ubuntu 19.10 setup ...

**Comment by lonvia** [comment]
Works for me with postgresql 12 + postgis 3 + SQLAlchemy 1.3.18 and just the 4 relations. Just to be sure that it is not something else, could you quickly check if the [test file](http://kilda.lonvia.de/test.pbf) file imports for you?

**Comment by hholzgra** [comment]
test file works fine

**Comment by lonvia** [comment]
Ok, thanks. That is mildly annoying. I'll try with a planet then.

**Comment by lonvia** [comment]
There are a couple of other issues with the new `waymarkedtrails-backedend` repo but computation of segments went fine and I see the offending segment in the final tables. So we can exclude Postgis 3 or SQLAlchemy as the culprit here. I'm out of idea now. If you can reproduce the issue with something significantly smaller than a planet file, I could spin up a Ubuntu 20 in a VM an have a look. I don't have a machine that is able to import a planet.

**Comment by hholzgra** [comment]
This *may* have been a false alarm after all ... not sure what I did different this time, and it is not final as the hiking import is still running, but it has been running for almost 10 minutes in the segment step right now, while previously it always failed within less than a minute when getting there ...

**Comment by hholzgra** [comment]
Different problem now:

```
20-06-28 14:42:44 Importing changed_objects...
20-06-28 14:42:44 Importing filtered_relations...
20-06-28 14:42:47 Importing way_relations...
20-06-28 14:42:51 Using 1 parallel threads.
20-06-28 15:06:18 Importing segments...
20-06-28 15:06:59 Using 1 parallel threads.
20-06-28 15:17:51 Importing hierarchy...
20-06-28 15:17:52 Importing routes...
20-06-28 15:17:52 Using 1 parallel threads.
Exception in thread Thread-3:
Traceback (most recent call last):
  File "<frozen zipimport>", line 177, in get_data
KeyError: 'wmt_shields/data/osmc/hiker.svg'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.8/dist-packages/osgende/common/threads.py", line 143, in loop
    self.process_func(req)
  File "/home/maposmatic/styles/waymarkedtrails-backend/db/tables/routes.py", line 200, in _process_construct_next
    cols = self._construct_row(obj, self.thread.conn)
  File "/home/maposmatic/styles/waymarkedtrails-backend/db/tables/routes.py", line 283, in _construct_row
    sym.to_file(os.path.join(self.config.symbol_datadir,
  File "/usr/local/lib/python3.8/dist-packages/waymarkedtrails_shields-0.1-py3.8.egg/wmt_shields/common/shield_maker.py", line 82, in to_file
  File "/usr/local/lib/python3.8/dist-packages/waymarkedtrails_shields-0.1-py3.8.egg/wmt_shields/common/shield_maker.py", line 104, in create_image
  File "/usr/local/lib/python3.8/dist-packages/waymarkedtrails_shields-0.1-py3.8.egg/wmt_shields/styles/osmc_symbol.py", line 95, in render
  File "/usr/local/lib/python3.8/dist-packages/waymarkedtrails_shields-0.1-py3.8.egg/wmt_shields/styles/osmc_symbol.py", line 369, in paint_fg_hiker
  File "/usr/local/lib/python3.8/dist-packages/waymarkedtrails_shields-0.1-py3.8.egg/wmt_shields/styles/osmc_symbol.py", line 376, in _src_from_svg
  File "/usr/local/lib/python3.8/dist-packages/waymarkedtrails_shields-0.1-py3.8.egg/wmt_shields/common/shield_maker.py", line 72, in find_resource
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 1157, in resource_string
    return get_provider(package_or_requirement).get_resource_string(
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 1402, in get_resource_string
    return self._get(self._fn(self.module_path, resource_name))
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 1570, in _get
    return self.loader.get_data(path)
  File "<frozen zipimport>", line 179, in get_data
OSError: [Errno 0] : 'wmt_shields/data/osmc/hiker.svg'
20-06-28 15:17:54 Internal error. Thread died. Killing other threads.
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/osgende/common/threads.py", line 73, in add_task
    self.queue.put(data, True, 2)
  File "/usr/lib/python3.8/queue.py", line 147, in put
    raise Full
queue.Full

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./makedb.py", line 210, in <module>
    exit(action())
  File "./makedb.py", line 104, in construct
    self.mapdb.construct()
  File "/usr/local/lib/python3.8/dist-packages/osgende/mapdb.py", line 91, in construct
    tab.construct(self.engine)
  File "/home/maposmatic/styles/waymarkedtrails-backend/db/tables/routes.py", line 150, in construct
    self.insert_objects(engine, subset)
  File "/home/maposmatic/styles/waymarkedtrails-backend/db/tables/routes.py", line 194, in insert_objects
    workers.add_task(obj)
  File "/usr/local/lib/python3.8/dist-packages/osgende/common/threads.py", line 81, in add_task
    raise Exception("Internal error. Thread died.")
Exception: Internal error. Thread died.
20-06-28 15:17:54 Used memory by index: 60967108272

```

**Comment by hholzgra** [comment]
Ok, that seems to be a problem with waymarkedtrails-shields packaging:

```
/usr/local/lib/python3.8/dist-packages # unzip -l waymarkedtrails_shields-0.1-py3.8.egg  | grep svg
```
Opening a new issue over there, closing this one ...

**Comment by lonvia** [comment]
The inclusion of data files for waymarkedtrails-shields does not seem to work recursively. Hopefully fixed with the latest commit.

**Comment by hholzgra** [comment]
Looks as if '**' for recursive inclusion does not actually work in setup.py patterns? I couldn't find anything documenting details on the reference level about this though ...
